### DISCUSSION TITLE
JS Plugin  - Broken Pipe Error - Large Payload Size - 223.86 KB Size

### DISCUSSION THREAD

**Discussion by pugazhendhit** [Kong/kong#7629] [Help] [answered]
When sending the http request with large payload size , getting broken pipe error in JS plugin , when reading the request payload

Postman ---------------------> Kong Gateway [ JS Plugin that decrypt the request ] ------------------------- > Micro Service

Kong version
kong-pdk@0.3.4
kong:2.4.1-alpine

Code :
if (!this.isWhiteList) {
let requestRaw = await kong.request.getRawBody() ------> exception here

Logs
2021/07/29 20:43:57 [notice] 44#0: *234 [kong] mp_rpc.lua:286 [config-redis] [ApiWhiteListConfig] is white list: false, client: 172.23.0.1, server: kong, request: "POST /paotang/v1/registration/grant HTTP/1.1", host: "localhost:9000"
2021/07/29 20:43:57 [error] 44#0: *234 send() failed (32: Broken pipe), client: 172.23.0.1, server: kong, request: "POST /paotang/v1/registration/grant HTTP/1.1", host: "localhost:9000"
2021/07/29 20:43:57 [notice] 43#0: signal 17 (SIGCHLD) received from 51
2021/07/29 20:43:57 [error] 44#0: *234 [kong] mp_rpc.lua:308 [decrypt] broken pipe, client: 172.23.0.1, server: kong, request: "POST /paotang/v1/registration/grant HTTP/1.1", host: "localhost:9000"
*2021/07/29 20:43:57 [notice] 43#0: 18 [kong] process.lua:258 external pluginserver 'js' terminated: exit 1, context: ngx.timer
2021/07/29 20:43:57 [notice] 43#0: *18 [kong] process.lua:248 Starting js, context: ngx.timer

**Comment by fffonion** [comment, marked as answer]
Duplicate of https://github.com/Kong/kong/issues/7628, will answer in the issue instead.
